packages/react/src/analytics/integrations/AnalyticsApi/AnalyticsApi.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  type AnalyticsApiIntegrationOptions,
+  type AnalyticsApiTrackData,
+} from './types/types.js';
+import { client } from '@farfetch/blackout-client';
+import {
+  type EventData,
+  integrations,
+  type LoadIntegrationEventData,
+  type StrippedDownAnalytics,
+  type TrackTypesValues,
+  utils,
+} from '@farfetch/blackout-analytics';
+import { isArray } from 'lodash';
+
+export default class AnalyticsAPI extends integrations.Integration<AnalyticsApiIntegrationOptions> {
+  static override [utils.CONSENT_CATEGORIES_PROPERTY] =
+    utils.DefaultConsentKeys.MARKETING;
+  private debugMode: boolean | undefined;
+
+  /**
+   * Creates an instance of Analytics Api integration.
+   *
+   * @param options               - User configured options.
+   * @param loadData              - Analytics's load event data.
+   * @param strippedDownAnalytics - Analytics stripped down instance.
+   */
+  constructor(
+    options: AnalyticsApiIntegrationOptions,
+    loadData: LoadIntegrationEventData,
+    strippedDownAnalytics: StrippedDownAnalytics,
+  ) {
+    super(options, loadData, strippedDownAnalytics);
+
+    // Check if options don't exists
+    if (!options) {
+      throw new Error(
+        '[Analytics API] - Analytics Api integration options must be set.',
+      );
+    }
+
+    if (!isArray(options.conversionsAPI?.eventsList)) {
+      throw new Error(
+        '[Analytics API] - The value `conversionsAPI.eventsList` from Analytics Api integration options must be an array.',
+      );
+    }
+
+    this.debugMode =
+      utils.getCookie('analyticsAPIDebug')?.toLowerCase() === 'true';
+  }
+
+  /**
+   * Track method that will call Analytics API with the necessary data per event.
+   * If there's more than one pageType per event, it will call Analytics API with each page type at a time.
+   *
+   * @param data - Track event data.
+   *
+   * @returns Promise that will resolve when the method finishes.
+   */
+  override async track(data: EventData<TrackTypesValues>) {
+    if (this.debugMode) {
+      utils.logger.info('ANALYTICS API track:', {
+        data,
+        options: this.getOptions(),
+      });
+    }
+
+    if (
+      (this.options.conversionsAPI.eventsList as Array<string>).includes(
+        data.event,
+      )
+    ) {
+      return await this.callAPI({
+        data,
+        options: this.getOptions(),
+      });
+    }
+  }
+
+  /**
+   * Calls the Analytics API with the given payload.
+   *
+   * @param data - Analytics' event data.
+   */
+  async callAPI(data: AnalyticsApiTrackData) {
+    await client.post('/analytics/v1/events', data);
+  }
+
+  /**
+   * Obtain options with additional common properties.
+   *
+   * @returns The options data.
+   */
+  private getOptions() {
+    return { ...this.options, debugMode: this.debugMode };
```

Comment:
Question: Why is it necessary to send the `debugMode` property to the server as it is derived from a cookie? Wouldn't the server receive the cookie as well and derive the `debugMode` from it?

Reply:
Debug mode, will be used for two things: first it's for log events on frontend layer.
The second it's related with AnalyticsApi server, witch are prepared to receive events, then return 200 (performance). Debug Mode flag, it's a flag added to await for internal responses and return more data. 

The point here isn't to set debug property on server side, and passed by frontend, that allow an easy way of setting debug only for developers or testers when they need that.

Reply:
I am not sure if I understood your comment. Debug mode is being set by the integration if the cookie `analyticsAPIDebug` is set. If the cookie is set, then the server will receive it when the POST request is made so it is redundant to send the `debugMode` in the payload of the POST request.

Reply:
This debugMode parameter is received from a cookie but needs to be included in the request body since the server side is ready to answer to this parameter through the body

Reply:
So the body of the request has to contain the `debugMode` parameter inside the `options` parameter or else the server will not assume it should treat the request in `debugMode` even though it receives the same cookie with it? Is that correct?

Reply:
In the AnalyticsAPI, we receive a single parameter, with data (the event data itself) and options inside. DebugMode is one of this properties that would be send in data.

Reply:
After the changes we discussed yesterday you should not send the options passed by the user to the server as it is unnecessary. Only pass the debugMode property.

Reply:
I don't remember of discarding options, and if it's to remove options, it will be need to change service that is already in live. I think we should maintain options, avoiding create breaking changes.